akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -546,6 +546,7 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
           updateLastSequenceNr(p)
           try {
             peekApplyHandler(p.payload)
+            onPersistSuccess(p.payload)
```

Comment:
Yeah, I think you're right. Would like to include a test for this case then - to make sure the onPersistFailure is properly invoked (I expect it to be, but if it's designed in behaviour, it needs a test :-))